tasks/cephfs/filesystem.py

```diff
@@ -436,25 +450,85 @@ def list_dirfrag(self, dir_ino):
 
         # NB we could alternatively use librados pybindings for this, but it's a one-liner
         # using the `rados` CLI
-        args = ["rados", "-p", self.get_metadata_pool_name(), "listomapkeys", dirfrag_obj_name]
+        args = ["rados", "-p", pool] + args
+        p = remote.run(
+            args=args,
+            stdout=StringIO())
+        return p.stdout.getvalue().strip()
+
+    def list_dirfrag(self, dir_ino):
+        """
+        Read the named object and return the list of omap keys
+
+        :return a list of 0 or more strings
+        """
+
+        dirfrag_obj_name = "{0:x}.00000000".format(dir_ino)
+
         try:
-            p = remote.run(
-                args=args,
-                stdout=StringIO())
+            key_list_str = self.rados(["listomapkeys", dirfrag_obj_name])
         except CommandFailedError as e:
             log.error(e.__str__())
             raise ObjectNotFound(dirfrag_obj_name)
 
-        key_list_str = p.stdout.getvalue().strip()
         return key_list_str.split("\n") if key_list_str else []
 
-    def journal_tool(self, args):
+    def erase_metadata_objects(self, prefix):
         """
-        Invoke cephfs-journal-tool with the passed arguments, and return its stdout
+        For all objects in the metadata pool matching the prefix,
+        erase them.
+
+        This O(N) with the number of objects in the pool, so only suitable
+        for use on toy test filesystems.
+        """
+        all_objects = self.rados(["ls"]).split("\n")
+        matching_objects = [o for o in all_objects if o.startswith(prefix)]
+        for o in matching_objects:
+            self.rados(["rm", o])
+
+    def erase_mds_objects(self, rank):
         """
+        Erase all the per-MDS objects for a particular rank.  This includes
+        inotable, sessiontable, journal
+        """
+        self.erase_metadata_objects("20{rank}.".format(rank=rank))
+        self.erase_metadata_objects("30{rank}.".format(rank=rank))
+        self.erase_metadata_objects("40{rank}.".format(rank=rank))
+        self.erase_metadata_objects("mds{rank}_".format(rank=rank))
```

Comment:
If I remember how these map correctly, this will break if get ever get 10 MDSes in a teuthology cluster?

Shouldn't be a problem, but please document it.